core/util_test.go

```diff
@@ -120,3 +122,37 @@ func TestUniqueLowerNames(t *testing.T) {
 	sort.Strings(u)
 	test.AssertDeepEquals(t, []string{"bar.com", "baz.com", "foobar.com"}, u)
 }
+
+// FIXME move to core, i guess
```

Comment:
Resolve this FIXME?

Also, I think the test cases below should probably check the Type field too, right?